validator_client/signing_method/src/lib.rs

```diff
@@ -227,11 +232,7 @@ impl SigningMethod {
 
                 // Determine the Web3Signer message type.
                 let message_type = object.message_type();
-
-                if matches!(
-                    object,
-                    Web3SignerObject::Deposit { .. } | Web3SignerObject::ValidatorRegistration(_)
-                ) && fork_info.is_some()
+                if matches!(message_type, MessageType::ValidatorRegistration) && fork_info.is_some()
```

Comment:
`object` can never be a `Web3SignerObject::Deposit` as a `SignableMessage` doesn't convert to it, see line `202` above

Reply:
Feels kinda hard to keep track if `SignableMessage` starts converting to a `Web3SignerObject::Deposit` in the future. 
I don't see the issue with keeping it as is, but what you said makes sense as well.
Although, i don't have enough context on the web3signer to know for sure